components/sidebar/sidebar_service.cc

```diff
@@ -335,7 +407,15 @@ std::vector<SidebarItem> SidebarService::GetHiddenDefaultSidebarItems() const {
           return default_item.built_in_item_type ==
                  added_item.built_in_item_type;
         });
-    default_items.erase(iter);
+    DCHECK(iter != default_items.end());
```

Comment:
It is against the chromium style guide to have a DCHECK and a check just after. Please remove this check, or the conditional if after.

Reply:
I also think using this DCHECK alone would be fine.

Reply:
Notice that I have a different PR that makes these changes here unnecessary, and I plan to merge it before this PR.

Reply:
I have rebased this PR on your PR now @cdesouza-chromium. Pushing in a minute